docs/content/concepts/resources.mdx

```diff
@@ -111,6 +111,39 @@ def db_resource(init_context):
     return DatabaseConnection(connection)
 ```
 
+## Context Manager Resources
+
+Dagster resources can serve as context managers, for scenarios where it is necessary to perform some sort of cleanup of the resource after execution. Let’s take the example of a database connection. We might want to clean up the connection once we are done using it. We can incorporate this into our resource like so:
+
+```python file=/concepts/resources/resources.py startafter=start_cm_resource endbefore=end_cm_resource
+from contextlib import contextmanager
+from dagster import resource
+
+
+@resource
+@contextmanager
+def my_cm_resource(_):
+    yield "foo"
```

Comment:
should this have a try/finally section?  do you need to run the snapshot again?

Reply:
It would be more realistic if it did, yea. Probably best if I just reuse the resource.